editoast/schemas/src/rolling_stock/supported_signaling_system.rs

```diff
@@ -0,0 +1,28 @@
+use serde::Deserialize;
+use serde::Serialize;
+use strum::Display;
+use strum::EnumString;
+use strum::FromRepr;
+use utoipa::ToSchema;
+
+use crate::rolling_stock::EtcsBrakeParams;
+
+#[derive(
+    Clone, Debug, PartialEq, Deserialize, Serialize, Display, EnumString, FromRepr, ToSchema,
+)]
+pub enum SignalingSystem {
+    BAL,
+    BAPR,
+    TVM300,
+    TVM430,
+}
+
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize, Display, ToSchema)]
+#[serde(untagged)]
+#[allow(clippy::large_enum_variant)]
+pub enum RollingStockSupportedSignalingSystem {
+    #[strum(to_string = "{0}")]
+    Simple(SignalingSystem),
+    #[strum(to_string = "ETCS_LEVEL2")]
+    EtcsLevel2 { brake_params: EtcsBrakeParams },
+}
```

Comment:
Why do we need a level of indirection with `Simple`? Can't we just simply do the following?
```rust
pub enum RollingStockSupportedSignalingSystem {
    BAL,
    BAPR,
    TVM300,
    TVM430,
    #[strum(to_string = "ETCS_LEVEL2")]
    EtcsLevel2 { brake_params: EtcsBrakeParams },
}
```

Reply:
It’s to make it work with the untagged enum, but if we switch to a tagged one, I can do that.